### ISSUE TITLE
New version doesn't compile on android

### ISSUE THREAD

**Issue by frederikstonge** [juliansteenbakker/mobile_scanner#1051]
I updated from 5.0.1 to 5.0.2 to fix the torch, but my builds are now failing for android : 

e: /home/vsts/.pub-cache/hosted/pub.dev/mobile_scanner-5.0.2/android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt: (171, 61): Unresolved reference: entries
e: /home/vsts/.pub-cache/hosted/pub.dev/mobile_scanner-5.0.2/android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt: (171, 77): Unresolved reference: it

**Comment by MishaKomar** [comment]
Same

**Comment by gigafunk** [comment]
Yeah I think the use of gradle 8.4 in this might be causing a problem. flutter 3.22, curerent main branch only supports up to gradle 7.6.1. They bumped it from 8.2 which was working, to 8.4 which is not. But flutter still only supports 7.6.1 so unless that gradle bump was for a functional reason, maybe we can get it bumped back to 8.2? or maybe even 7.6.1 to be in compliance with flutter norms

**Comment by joshuariveramnltech** [comment]
same issue here, changed my version  to  [`5.0.0-beta.3`](https://github.com/juliansteenbakker/mobile_scanner/releases/tag/v5.0.0-beta.3) and is able to build fine.

[`^5.0.0`](https://github.com/juliansteenbakker/mobile_scanner/releases/tag/v5.0.0-beta.3) seems to cause the issue.
Any idea how to fix this?

**Comment by arnabonetraker** [comment]
5.0.2 has this issue.
For now reverting back to 5.0.1 works

```This declaration needs opt-in. Its usage must be marked with '@kotlin.ExperimentalStdlibApi' or '@OptIn(kotlin.ExperimentalStdlibApi::class)'```

**Comment by datpt11** [comment]
same issue

**Comment by juliansteenbakker** [comment]
Hi all, i have a possible fix for this on the master branch of this repository. If anyone can verify that this is working i can release a new version. Thanks!

**Comment by EArminjon** [comment]
Master compile well, please release :') !

I tested this (I took latest commit)
```yaml
  mobile_scanner:
    git:
      url: https://github.com/juliansteenbakker/mobile_scanner.git
      ref: 695530aed11a7a0401ffcf7b3a80ec0e0532ea11
```

**Comment by navaronbracke** [comment]
@juliansteenbakker The upgrade of the Kotlin dependency must have broken things in 5.0.2 it seems (the `entries` declaration is available only for newer Kotlin version apparently)

Can we provide the `onDetect` / `autoStart` parameters specifically with a named constructor on the widget? Then we can assert that the arguments are not mixed up when a controller is provided.

**Comment by navaronbracke** [comment]
@juliansteenbakker I did a quick review and I have some feedback to make sure that the reinstated version won't cause regressions relating to how the functionality was brought back.

See my comments here: https://github.com/juliansteenbakker/mobile_scanner/commit/32d670da56ccc0afd66a73e901fddb565d0b6694

**Comment by juliansteenbakker** [comment]
> Can we provide the onDetect / autoStart parameters specifically with a named constructor on the MobileScanner widget? Then we can assert that the arguments are not mixed up when a controller is provided. Or perhaps just with an assert?

I think its best to let the "old" not named paremeter in place so its not breaking, and indeed use an assert to check if the usage is not mixed up. I'm not against changes in the parameters, however i think we should have a deprecation marker first before (re)moving parameters so users know how to update.

**Comment by navaronbracke** [comment]
Indeed, I had only added the note in the changelog, but had forgotten to provide the `deprecated` annotation.

I will provide a pull request to address the small remarks that I had with the brought back API. You can expect these changes in the afternoon.

**Comment by A-bahattab** [comment]
same issue here

**Comment by juliansteenbakker** [comment]
I have released v5.1.0 which should solve this issue.
